### DISCUSSION TITLE
Create node with attributes with add_edge()

### DISCUSSION THREAD

**Discussion by FloLaco** [networkx/networkx#4654] [Q&A] [answered]
Hello,

I'm try to build a graph dynamicaly and I would like to summurize my lines.
Actually I have, and it's working :
```
graph.add_node(neighbor['hostname'], device=neighbor_node)
graph.add_edge(str(current_node), (neighbor['hostname'])
```
`current_node` is already existing in my Graph but not my `neighbor['hostname']`
Now I'm trying to remove the `add_node` line because the `add_edge` is adding the node if it does not exist, but how to embeded attributes for my node (not for my edge).

I try this thing but it's not working 

```
graph.add_edge(str(current_node), (neighbor['hostname'], {'device': neighbor_node}))
```

Thanks

**Comment by dschult** [comment, marked as answer]
You will need a second line to add the node attribute. 
There isn't a one-line way to add an edge and also assign attributes to one of the nodes.
